blocks/components/editable/tinymce.js

```diff
@@ -19,12 +22,42 @@ export default class TinyMCE extends wp.element.Component {
 		}
 	}
 
-	shouldComponentUpdate() {
+	componentDidMount() {
+		this.initialize();
+	}
+
+	shouldComponentUpdate( nextProps ) {
 		// We must prevent rerenders because TinyMCE will modify the DOM, thus
 		// breaking React's ability to reconcile changes.
 		//
 		// See: https://github.com/facebook/react/issues/6802
-		return false;
+		return nextProps.tagName !== this.props.tagName;
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( ! this.editor ) {
+			return;
+		}
+
+		if ( nextProps.tagName !== this.props.tagName ) {
+			this.editor.destroy();
```

Comment:
>The node and contents will still be there. Is that was you mean?

It's what I was curious about, yes, but opposite of the assumption I was making. In #635 I'd encountered issues where React would reconcile based on the DOM that TinyMCE had since modified, resulting in an infinite loop of errors, particularly around splitting and merging blocks. With these changes I'm not seeing this issue, but I don't feel confident in knowing why it's not a problem.